bindings/gcp/bucket/metadata.yaml

```diff
@@ -0,0 +1,106 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: gcp.bucket
+version: v1
+status: alpha
+title: "GCP Storage Bucket"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/gcpbucket/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create an item."
+capabilities: []
+authenticationProfiles:
+  - title: "GCP API call Authentication"
+    description: |
+      Authenticate authenticates API calls with the given service account or refresh token JSON credentials.
+    metadata:
+      - name: private_key_id
+        required: true
+        sensitive: true
+        description: |
+          GCP private key id.
+        example: '"privateKeyId"'
+      - name: private_key
+        required: true
+        sensitive: true
+        description: |
+          GCP credentials private key. Replace with x509 cert.
+        example: '"12345-12345"'
+metadata:
+  - name: bucket
+    required: true
+    description: |
+      The bucket name.
+    example: '"mybucket"'
+    type: string
+  - name: type
+    type: string
+    required: true
+    description: |
+      The GCP credentials type.
+    example: '"service_account"'
```

Comment:
What are the other options? We have an enum type that allows listing all options

Reply:
@ItalyPaleAle Based on the way the component code is setup, it only supports `service_account`.

Reply:
Do we need this option then?

Reply:
I moved all the authentication related fields to the built-in Authentication profile for `gcp` as this plus other fields/properties go together as part of the GCP authentication with a service account.